src/cmd/flux-job.c

```diff
@@ -1158,6 +1185,124 @@ void attach_stdin_cb (flux_reactor_t *r, flux_watcher_t *w,
     }
 }
 
+void MPIR_Breakpoint ()
+{
+    /* FIXME: Add this so that the compiler won't do funky things
+     * to this dummy function, in which case the debugger
+     * might have hard time setting a breakpoint here
+     */
+    printf ("MPIR_Breakpoint() is called\n");
+    printf ("MPIR_Breakpoint() is about to return\n");
+}
+
+static void valid_or_exit_for_debug (struct attach_ctx *ctx)
+{
+    int priority = 0;
+    uint32_t userid = 0;
+    flux_jobid_t id = 0;
+    double t_submit = 0.0f;
+    flux_future_t *f = NULL;
+    flux_job_state_t state = FLUX_JOB_INACTIVE;
+
+    if (!(f = flux_rpc_pack (ctx->h,
+                             "job-manager.info",
+                             FLUX_NODEID_ANY,
+                             0,
+                             "{s:I}",
+                             "id",
+                             ctx->id)))
+        log_err_exit ("flux_rpc_pack");
+
+    if (flux_rpc_get_unpack (f, "{s:I s:i s:i s:f s:i}",
+                                "id",
+                                &id,
+                                "userid",
+                                &userid,
+                                "priority",
+                                &priority,
+                                "t_submit",
+                                &t_submit,
+                                "state",
+                                &state) < 0)
```

Comment:
Style: for pack/unpack, we seem to have settled on a style that collects key, value on the same line, e.g.:
```C
    if (flux_rpc_get_unpack (f, "{s:I s:i s:i s:f s:i}",
￼                                "id", &id,
￼                                "userid", &userid,
￼                                "priority", &priority,
￼                                "t_submit", &t_submit,
￼                                "state", &state) < 0)
```

However, that's not official so probably final style is up to you.

Edit: Same comment for `flux_rpc_pack()` above.

Reply:
Sure. I can make that change. This may change in the future when this will get this from `job-info`. But it would hurt to do this now as a reminder.

Reply:
Yeah, that was just meant as an FYI :wink:

Reply:
Yeah but this FYI is good!

I think it makes sense to change it (as well as fetching only state) now so that when `job-info` comes hopefully we can make one change and we can land this.  :-)